### ISSUE TITLE
SharpPcap Linux DotNetCore 3.0 Buffering Frames

### ISSUE THREAD

**Issue by APM3** [chmorgan/sharppcap#90]
I'm doing the basic Promiscuous packet capture on Linux using OnPacketArrival and it appears that the frames are getting buffered for a good bit of time, several hundred milliseconds.  I also use a standard UDP Listener for different traffic, and the time difference is noticeable.  Is this expected?

12/12/2019 21:44:38 -   12/13/2019 02:44:38.780063 UDP Listener Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.110072 UDP Listener Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.440094 UDP Listener Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.770370 UDP Listener Record
12/12/2019 21:44:39 -   12/13/2019 02:44:38.776343 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:38.807358 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:38.838638 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:38.869509 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:38.900561 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:38.931643 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:38.962662 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:38.993734 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.024777 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.055795 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.086889 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.117931 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.149061 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.180072 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.211091 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.242206 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.273242 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.304271 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.335307 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.366380 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.397433 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.428521 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.459608 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.490597 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.521642 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.552682 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.583712 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.614833 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.645923 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.676909 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.707930 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.739011 SharpPcap Record
12/12/2019 21:44:39 -   12/13/2019 02:44:39.769986 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.100111 UDP Listener Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.430382 UDP Listener Record
12/12/2019 21:44:40 -   12/13/2019 02:44:39.801121 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:39.832145 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:39.863232 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:39.894280 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:39.925306 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:39.956400 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:39.987541 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.018481 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.049582 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.080576 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.111721 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.142774 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.173760 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.204880 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.235939 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.266999 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.297979 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.329037 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.360102 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.391080 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.422184 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.453209 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.484322 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.515441 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.546441 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.577475 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.608580 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.639559 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.670695 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.701759 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.732829 SharpPcap Record
12/12/2019 21:44:40 -   12/13/2019 02:44:40.763825 SharpPcap Record

**Comment by chmorgan** [comment]
I'm surprised you would see a large amount of time between when a packet came in and when it was received. There is a chance that your printing is affecting your results though. Are you storing this timing data to a file for display afterwards? What kind of system is this running on?

**Comment by APM3** [comment]
Sorry Chris. I figured it out by reading the code. The “problem” was that
the default read timeout is one second. Once I changed that property
according to my specific expected parameters, I receive my packets
promptly.

For what its worth though, the inner time in the log was the time received
via sharppcap and the outer time was the time that my app received it in
the PacketReceived callback, and put it in my print queue.

Thanks,
